spec/features/user_edits_book_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+feature 'User edits an existing book' do
+  scenario 'updates title field successfully' do
+    @book = create(:book)
+    visit root_path
+
+    edit_book(@book, new_title: 'The Golden Compass',
+                     new_author: @book.author)
+
+    expect(page.current_path).to eq books_path
+  end
+
+  scenario 'updates title and author fields successfully' do
+    @book = create(:book)
+    visit root_path
+
+    edit_book(@book, new_title: 'The Amber Spyglass',
+                     new_author: 'Philip Pullman')
+
+    expect(page.current_path).to eq books_path
+  end
+
+  def edit_book(book, options={})
```

Comment:
Good question! There are a couple things I would do. You can remove the `book` argument and click the edit link, you shouldn't need to explicitly `visit` the edit page since you clicked on it in the test.

Then you could use keyword arguments to accept different arguments and give defaults if you want: https://robots.thoughtbot.com/ruby-2-keyword-arguments from Ian is fantastic to give you a primer on keywords args